crypto/Makefile

```diff
@@ -3,49 +3,104 @@ COVER_PROFILE := cover.out
-.PHONY: non_relic_tests
-non_relic_tests:
-# root package without relic 
-	go test -coverprofile=$(COVER_PROFILE) $(RACE_FLAG) $(if $(JSON_OUTPUT),-json,) $(if $(NUM_RUNS),-count $(NUM_RUNS),) $(if $(VERBOSE),-v,)
-# sub packages
-	go test -coverprofile=$(COVER_PROFILE) $(RACE_FLAG) $(if $(JSON_OUTPUT),-json,) $(if $(NUM_RUNS),-count $(NUM_RUNS),) $(if $(VERBOSE),-v,) ./hash
-	go test -coverprofile=$(COVER_PROFILE) $(RACE_FLAG) $(if $(JSON_OUTPUT),-json,) $(if $(NUM_RUNS),-count $(NUM_RUNS),) $(if $(VERBOSE),-v,) ./random
+# format C code
+.PHONY: c-format
+c-format:
+	clang-format -style=llvm -dump-config > .clang-format
+	clang-format -i *.c
+	clang-format -i *.h
+	rm -f .clang-format
+	git diff --exit-code
 
-############################################################################################
-# CAUTION: DO NOT MODIFY THIS TARGET! DOING SO WILL BREAK THE FLAKY TEST MONITOR
+# address sanitization and other checks
+.SILENT: c-asan
+c-asan:
+# - address sanitization and other checks (only on linux)
+	if [ $(UNAME) = "Linux" ]; then \
+		$(CGO_FLAG) CC="clang -O0 -g -fsanitize=address -fno-omit-frame-pointer -fsanitize=leak -fsanitize=undefined -fno-sanitize-recover=all -fsanitize=float-divide-by-zero -fsanitize=float-cast-overflow -fno-sanitize=null -fno-sanitize=alignment" \
+		LD="-fsanitize=address -fsanitize=leak" go test; \
+		if [ $$? -ne 0 ]; then exit 1; fi; \
+	else \
+		echo "sanitization is only supported on Linux"; \
+	fi; \
 
-# sets up the crypto module and runs all tests
-.PHONY: test
-test: setup unittest
+# memory sanitization
+.SILENT: c-msan
+c-msan:
+# - memory sanitization (only on linux and using clang) - (could use go test -msan)
+# currently, this leads to many false positives, most likely because of assembly code not handled properly
+# by asan. If you would like to run this command, you can use `NO_MSAN` to diable msan in some C functions.
+# For instance "void NO_MSAN f() {...}" disables msan in function f. `NO_MSAN` is already defined in
+# bls12381_utils.h
+	if [ $(UNAME) = "Linux" ]; then \
+		$(CGO_FLAG) CC="clang -DMSAN -O0 -g -fsanitize=memory -fno-omit-frame-pointer -fsanitize-memory-track-origins" \
+		LD="-fsanitize=memory" go test; \
+		if [ $$? -ne 0 ]; then exit 1; fi; \
+	else \
+		echo "sanitization is only supported on Linux"; \
+	fi; \
+
+# sanitize C code
+.SILENT: c-sanitize
+c-sanitize: c-asan
+# - address sanitization and other checks (only on linux)
+# - memory sanitization (target m-san) is disabled because of multiple false positives
 
-# runs the unit tests of the module (assumes the module was set up)
-.PHONY: unittest
-unittest: relic_tests non_relic_tests
 
-############################################################################################
+
+# Go tidy
+.PHONY: go-tidy
+go-tidy:
+	go mod tidy -v
+	git diff --exit-code
+
+# Go lint
+.PHONY: go-lint
+go-lint:
+lint: go-tidy
+	# revive -config revive.toml
+	golangci-lint run -v ./...
+	
+	
+
```

Comment:
Can we remove the extra new lines between target sections? This also applies to the extra newlines in lines 78-79. If needed, we can add a header / title before a section of targets so the grouping is clearer.